src/backends/tensorrt/plan_backend.cc

```diff
@@ -3044,6 +3047,17 @@ PlanBackend::Context::Run(
     nvinfer1::Dims dims;
     dims = citr->second.context_->getBindingDimensions(binding_index);
 
+    // Wait for model execution to process the output tensors with pinned memory
+    // address if zero-copy is supported. Otherwise use device memory address,
+    // so that the memory copies perform asynchronously and wait for model
+    // execution.
+    void* buffer = nullptr;
+    if (zero_copy_support_) {
+      buffer = io_binding_info.buffer_;
+    } else {
+      buffer = io_binding_info.device_buffer_;
```

Comment:
You don't need this, just pass `io_binding_info.buffer_` to responder's ProcessTensor

Reply:
Doesn't it require to be device_buffer_ for non iGPU case?

Reply:
See what `buffer_` is set to in regular case

Reply:
I see what you mean. Fixed it now